### ISSUE TITLE
After tokenizers upgrade, the length of the token does not correspond to the length of the model

### ISSUE THREAD

**Issue by CurtainRight** [huggingface/transformers#36574]
### System Info

#36532 
1. My version is 4.48.1, a relatively new version. After referring to the document and executing it, my reasoning is still abnormal and the result is the same as the original reasoning
`import torch
import transformers
from transformers import PegasusForConditionalGeneration

# 加载 Pegasus 模型
#model = PegasusForConditionalGeneration.from_pretrained('google/pegasus-large')

# 获取模型参数
params = model.state_dict()
print(params)

# 获取词嵌入层权重（Pegasus 的词嵌入层是 `model.shared.weight`）
embeddings = params['model.shared.weight']
print(embeddings)

# 计算扩展前的词嵌入统计量
pre_expansion_embeddings = embeddings[:-3, :]  # 排除最后 3 个 token
mu = torch.mean(pre_expansion_embeddings, dim=0)  # 均值
n = pre_expansion_embeddings.size()[0]  # 词嵌入数量
sigma = ((pre_expansion_embeddings - mu).T @ (pre_expansion_embeddings - mu)) / n  # 协方差矩阵

# 定义多元正态分布
dist = torch.distributions.multivariate_normal.MultivariateNormal(
    mu, covariance_matrix=1e-5 * sigma
)

# 生成新的词嵌入
new_embeddings = torch.stack(tuple((dist.sample() for _ in range(3))), dim=0)
print(new_embeddings)

# 替换词嵌入层的最后 3 个 token
embeddings[-3:, :] = new_embeddings
params['model.shared.weight'][-3:, :] = new_embeddings

# 加载更新后的参数
model.load_state_dict(params)`
3. And I found that on NPU, although the token has changed, there are no errors and the reasoning is normal. Transformers and tokenizers are the same, only the torch version is different. On NPU, it is 2.1.0

### Who can help?

hi,@Rocketknight1 
Please don't close it yet, how can I adjust and adapt to the new version.thanks

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

Please don't close it yet, how can I adjust and adapt to the new version.thanks

### Expected behavior

Please don't close it yet, how can I adjust and adapt to the new version.thanks
